riscv-smtdeleg-sstcfg.adoc

```diff
@@ -0,0 +1,98 @@
+[[intro]]
+== Introduction
+
+For a Machine-level environment, extension *Smtdeleg* (‘Sm’ for Privileged architecture and Machine-level extension, ‘tdeleg’ for Trigger Delegation) encompasses all added CSRs and all behavior modifications for a hart, over all privilege levels. For a
+Supervisor-level environment, extension *Sstcfg* (‘Ss’ for Privileged architecture and Supervisor-level extension, ‘tcfg’ for Trigger Configuration) provides access to delegated triggers. These extensions depend on the Sdtrig and Sscsrind extensions.
+
+[[body]]
+== CSRs
+
+=== Supervisor Trigger Select (`stselect`)
+
+`stselect` provides access to `tselect` from S-mode and VS-mode.
+
+== Supervisor and Virtual Supervisor Trigger Access
+
+While `siselect` holds XXX, the `sireg*` registers provide access to the `tdata*` and `tinfo` registers associated with the trigger selected in `tselect`.  The register mapping is shown below.
+
+.Indirect Trigger Register Mappings
+[width="50%",options="header"]
+|===
+| Indirect CSR | Sdtrig CSR 
+| `sireg` | `tdata1` 
+| `sireg2` | `tdata2` 
+| `sireg3` | `tdata3` 
+| `sireg4` | `tinfo` 
+|===
+
+[WARN]
+====
+_There has been much discussion of the best way to access counters in S-mode.  This method requires 1 new CSR (`stselect`), and creates a scenario where there are 2 levels of indirection (siselect routes sireg* to t*, and tselect routes t* to the state associated with the selected trigger)._
+
+_Also proposed was to simply allocate a range of `siselect` values that serve to select the trigger, bypassing `tselect` when accessing trigger state through `sireg*`.  This avoids the 2nd level of indirection, but artificially limits the number of triggers accessible in S-mode.  `tselect` supports up to 2^32^ triggers, while `siselect` would likely allocate only enough values to cover practical implementations (256?)._
+
+_A third approach was considered, akin to what is spec'd here but accessing `tselect` via `sireg5` instead of through a new `stselect` CSR.  While saving a CSR, this approach would require `sireg5` to continue to access `tselect` while the other `sireg*` registers shift the state they acccess based on the `tselect` value.  This approach seems more awkward for implementation._
+
+_Feedback on these options, or others, is welcome._
```

Comment:
I would prefer sdtrig to be mapped to Indirect CSRs with 2nd option, but it is too late. 
I prefer more compatible approach with the existing sdtrig and shadow register approach as 4th option: 
4th Option: Define stselect/stdata/stinfo as CSR not indirect CSRs (but I am not sure RISC-V community will be willing to spend CSR address space for them).

Reply:
Yes, given that Sscsrind was defined precisely to avoid needing new CSRs every time an extension needs an indirect access method, I doubt there will be enthusiasm for adding stdata* or stinfo.  In fact, I kind of suspect ARC will prefer my original proposal, without stselect.  But we'll see.  I can add your suggested option to the comment.

Reply:
Added, see revised text above.

Reply:
LGTM. thank you, resolving it.